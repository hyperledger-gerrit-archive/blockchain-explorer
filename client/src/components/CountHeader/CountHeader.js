import React, { Component } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';
import { withStyles } from 'material-ui/styles';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import Button from 'material-ui/Button';
import Tooltip from 'material-ui/Tooltip';
import Typography from 'material-ui/Typography';
import PropTypes from 'prop-types';
import TransactionsView from "../View/TransactionsView";
import ChaincodeView from "../View/ChaincodeView";
import { getHeaderCount as getCountHeaderCreator } from '../../store/actions/header/action-creators';

const styles = theme => ({
  card: { minWidth: 290, height: 100, },
  media: { height: 30, },
  title: {
    marginBottom: 16, fontSize: 16, color: theme.palette.text.secondary,
    position: 'absolute', right: 10, top: 10
  },
  pos: {
    marginBottom: 5,
    color: theme.palette.text.secondary,
    position: 'absolute',
    right: 10,
    top: 60,
    fontSize: 18,
  },

});

class CountHeader extends Component {
  constructor(props) {
    super(props);
    this.state = { isChaincodeView: true, isTransactionView: false };
 /*   this.state.countHeader = { 
        chaincodeCount: 0, txCount: 0, latestBlock: 0, peerCount: 0
     };
     */
    this.state = { countHeader: this.props.countHeader };
    this.handleClickChaincodeView = this.handleClickChaincodeView.bind(this);
    this.handleClickTransactionView = this.handleClickTransactionView.bind(this);
  }

  componentWillMount() {
    this.props.getCountHeader();    
    console.log('componentWillMount: ', this.state.countHeader);
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.countHeader !== this.props.countHeader) {
      this.setState({countHeader: nextProps.countHeader});
      console.log("countHeader received nextProps value: ", nextProps.countHeader);
    }
  }

  componentDidMount() {
  }

  componentDidUpdate(prevProps, prevState) {
    //TODO 
    // this will run in a loop, need to validate some state
    //Use componentDidUpdate if you want to update state asynchronously when props change!
    // only update  if the data has changed
      console.log('componentDidUpdate() ' + new Date().toLocaleString());
      if (prevState.countHeader !== prevProps.countHeader) {
        this.props.getCountHeader();
        console.log('componentDidUpdate() ' + new Date().toLocaleString());
        console.log('prevProps ', prevProps.countHeader, ' prevState ', prevState.countHeader);
      }
      
  }

  handleClickChaincodeView() {
    this.setState({ isTransactionView: false });
    this.setState({ isChaincodeView: true });
  }

  handleClickTransactionView() {
    this.setState({ isChaincodeView: false });
    this.setState({ isTransactionView: true });
  }


  render() {
    const { classes } = this.props;
    const { countHeader } = this.props.countHeader;
    
    let currentView = null;
    if (this.state.isChaincodeView) {
      currentView = <ChaincodeView />;
    } else if (this.state.isTransactionView) {
      currentView = <TransactionsView />;
    }

    return (
      <div>
        <div style={{ position: 'absolute', top: 100, left: 975, zIndex: 1000 }}>
          <Card className={classes.card}>
            <CardContent>
              <Typography className={classes.title}>CHAINCODE</Typography>
              <Typography className={classes.pos}>{countHeader.chaincodeCount}
              </Typography>
            </CardContent>
            <CardActions>
              <Tooltip id="tooltip-top" title="View Chaincode" placement="top">
                <Button color="primary" onClick={this.handleClickChaincodeView}>
                  More
          </Button>
              </Tooltip>
            </CardActions>
          </Card>
        </div>
        <div style={{ position: 'absolute', top: 100, left: 665, zIndex: 1000 }}>
          <Card className={classes.card}>
            <CardContent>
              <Typography className={classes.title}>TX</Typography>
              <Typography className={classes.pos}>{countHeader.txCount} </Typography>
            </CardContent>
            <CardActions>
              <Tooltip id="tooltip-top" title="View Transactions" placement="top">
                <Button color="primary" onClick={this.handleClickTransactionView}>
                  More
          </Button>
              </Tooltip>
            </CardActions>
          </Card>
        </div>
        <div style={{ position: 'absolute', top: 100, left: 355, zIndex: 1000 }}>
          <Card className={classes.card}>
            <CardContent>
              <Typography className={classes.title}>BLOCK</Typography>
              <Typography className={classes.pos}> {countHeader.latestBlock}</Typography>
            </CardContent>
          </Card>
        </div>
        <div style={{ position: 'absolute', top: 100, left: 50, zIndex: 1000 }}>
          <Card className={classes.card}>
            <CardContent>
              <Typography className={classes.title}>PEER</Typography>
              <Typography className={classes.pos}>{countHeader.peerCount}</Typography>
            </CardContent>
          </Card>
        </div>
        <div style={{ position: 'absolute', top: 210, left: 30, zIndex: 1000 }}>
          {currentView}
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getCountHeader: () => dispatch(getCountHeaderCreator())
});


const mapStateToProps = state => ({
  countHeader: state.countHeader
});

CountHeader.propTypes = {
  classes: PropTypes.object.isRequired,
  getCountHeader: PropTypes.func.isRequired,
};

export default compose(
  withStyles(styles, { name: 'CountHeader' }),
  connect(mapStateToProps, mapDispatchToProps),
)(CountHeader);

