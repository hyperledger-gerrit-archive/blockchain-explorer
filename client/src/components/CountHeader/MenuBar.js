import React, { Component } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';
import { withStyles } from 'material-ui/styles';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import { Badge } from 'reactstrap';
import Button from 'material-ui/Button';
import Tooltip from 'material-ui/Tooltip';
import Typography from 'material-ui/Typography';
import PropTypes from 'prop-types';
import Peers from '../Lists/Peers';
import Blocks from '../Lists/Blocks';
import Transactions from '../Lists/Transactions';
import DashboardView from '../View/DashboardView';
import Channels from '../Lists/Channels';
import { getBlockList as getBlockListCreator, getBlockInfo as getBlockInfoCreator } from '../../store/actions/block/action-creators';
import { getTransactionInfo as getTransactionInfoCreator } from '../../store/actions/transaction/action-creators';
import { getLatestBlock as getLatestBlockCreator } from '../../store/actions/latestBlock/action-creators';
import { getHeaderCount as getCountHeaderCreator } from '../../store/actions/header/action-creators';

import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap';

const styles = theme => ({
  card: { minWidth: 250, height: 100, },
  media: { height: 30, },
  title: {
    marginBottom: 16, fontSize: 16, color: theme.palette.text.secondary,
    position: 'absolute', right: 10, top: 10
  },
  pos: {
    marginBottom: 5,
    color: theme.palette.text.secondary,
    position: 'absolute',
    right: 10,
    top: 60,
    fontSize: 18,
  },

});

class MenuBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeView: 'DashboardView',
      activeTab: { dashboardTab: true, peersTab: false, blocksTab: false },
      countHeader: { countHeader: this.props.getCountHeader() }
    }

    this.handleClickTransactionView = this.handleClickTransactionView.bind(this);
    this.handleClickBlockView = this.handleClickBlockView.bind(this);
    this.handleClickChannelView = this.handleClickChannelView.bind(this);
    this.handleClickPeerView = this.handleClickPeerView.bind(this);
    this.handleClickDashboardView = this.handleClickDashboardView.bind(this);
  }

  componentWillMount() {

  }

  componentWillReceiveProps(nextProps) {
    if (JSON.stringify(nextProps.countHeader) !== JSON.stringify(this.props.countHeader)) {
      console.log('nextProps.countHeader !== this.props.countHeader')
      this.setState({ countHeader: nextProps.countHeader });
    }
  }


  componentDidMount() {

    //TODO isolate interval
    setInterval(() => {
      this.props.getCountHeader();
      this.props.getLatestBlock();

      //  console.log(this.props.countHeader.countHeader.latestBlock);
      // console.dir(this.props.blockList);
      // this.props.loadBlockList(this.props.countHeader.countHeader.latestBlock);
      this.props.getBlockList(this.props.countHeader.countHeader.latestBlock);
      // .then(data =>
      //   this.props.getBlockList(data.latestBlock)
      // ).catch(err => {
      //   // logger.error(err)
      // });
    }, 3000)

  }

  componentDidUpdate(prevProps, prevState) {
  }

  handleClickTransactionView() {
    this.setState({ activeView: 'TransactionView' });
  }
  handleClickBlockView() {
    this.setState({ activeView: 'BlockView' });
    this.setState({
      activeTab: {
        dashboardTab: false,
        peersTab: false,
        blocksTab: true
      }
    });
  }
  handleClickChannelView() {
    this.setState({ activeView: 'ChannelView' });
  }
  handleClickPeerView() {
    this.setState({ activeView: 'PeerView' });
    this.setState({
      activeTab: {
        dashboardTab: false,
        peersTab: true,
        blocksTab: false
      }
    });
  }
  handleClickDashboardView() {
    this.setState({ activeView: 'DashboardView' });
    this.setState({
      activeTab: {
        dashboardTab: true,
        peersTab: false,
        blocksTab: false
      }
    });
  }

  render() {
    const { classes } = this.props;
    const { countHeader } = this.props.countHeader;

    let currentView = null;
    /* could be routed */
    switch (this.state.activeView) {
      case 'TransactionView':
        currentView = <Transactions transactionList={this.props.transactionList} />;
        break;
      case 'BlockView':
        currentView = <Blocks blockList={this.props.blockList.rows}
          block={this.props.block}
          getBlockInfo={this.props.getBlockInfo}
          transaction={this.props.transaction}
          getTransactionInfo={this.props.getTransactionInfo} />;
        break;
      case 'ChannelView':
        currentView = <Channels channelList={this.props.channelList} />;
        break;
      case 'PeerView':
        currentView = <Peers peerList={this.props.peerList} />;
        break;
      case 'DashboardView':
        currentView = <DashboardView />;
        break;
      default:
        currentView = <DashboardView />;
        break;
    }

    return (
      <div>
        <div className="menuItems">
          <Navbar color="faded" light expand="md">
            <Nav className="ml-auto" navbar>
              <NavItem active={this.state.activeTab.dashboardTab} onClick={this.handleClickDashboardView}>DASHBOARD </NavItem>
              <NavItem active={this.state.activeTab.peersTab} onClick={this.handleClickPeerView}>NETWORK  </NavItem>
              <NavItem active={this.state.activeTab.blocksTab} onClick={this.handleClickBlockView}>BLOCKS </NavItem>
              <NavItem >TRANSACTIONS</NavItem>
              <NavItem >SMART CONTRACTS</NavItem>
            </Nav>
          </Navbar>
        </div>


        <div style={{ position: 'absolute', top: 210, left: 30, zIndex: 1000 }}>
          {currentView}
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getCountHeader: () => dispatch(getCountHeaderCreator()),
  getLatestBlock: () => dispatch(getLatestBlockCreator()),
  getBlockList: (latestBlock) => dispatch(getBlockListCreator(latestBlock)),
  getBlockInfo: (number) => dispatch(getBlockInfoCreator(number)),
  getTransactionInfo: (tx_id) => dispatch(getTransactionInfoCreator(tx_id))
});


const mapStateToProps = state => ({
  countHeader: state.countHeader,
  peerList: state.peerList.peerList,
  blockList: state.blockList.blockList,
  transactionList: state.transactionList.transactionList,
  channelList: state.channelList.channelList,
  block: state.block.block,
  transaction: state.transaction.transaction
});



export default compose(
  withStyles(styles, { name: 'CountHeader' }),
  connect(mapStateToProps, mapDispatchToProps),
)(MenuBar);
