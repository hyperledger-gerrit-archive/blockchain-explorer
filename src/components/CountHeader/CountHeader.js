import React, { Component } from 'react';

class CountHeader extends Component {
  render() {
    return (
    <div className="row" id="heads-up">
       Counter Header
    </div>

    );
  }
}
export default CountHeader;


