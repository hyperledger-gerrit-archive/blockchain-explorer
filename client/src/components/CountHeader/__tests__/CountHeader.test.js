import React from "react";
import { mount, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import CountHeader from "../CountHeader";

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {
    state: { isChaincodeView: true, isTransactionView: false },
};

describe('Testing CountHeader', () => {
    let wrapper;
    it('renders as expected', () => {
        wrapper = shallow(
            <CountHeader />,
            { context: { store: mockStore(initialState) } },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});