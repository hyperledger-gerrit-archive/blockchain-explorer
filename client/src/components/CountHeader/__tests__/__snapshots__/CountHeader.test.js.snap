// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CountHeader renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CountHeader)
    classes={
        Object {
            "card": "CountHeader-card-1",
            "media": "CountHeader-media-2",
            "pos": "CountHeader-pos-4",
            "title": "CountHeader-title-3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "card": "CountHeader-card-1",
        "media": "CountHeader-media-2",
        "pos": "CountHeader-pos-4",
        "title": "CountHeader-title-3",
      },
      "countHeader": undefined,
      "getCountHeader": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "card": "CountHeader-card-1",
          "media": "CountHeader-media-2",
          "pos": "CountHeader-pos-4",
          "title": "CountHeader-title-3",
        },
        "countHeader": undefined,
        "getCountHeader": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
